styx-api-service/src/main/java/com/spotify/styx/api/WorkflowResource.java

```diff
@@ -86,11 +94,61 @@ public WorkflowResource(Storage storage, DockerImageValidator dockerImageValidat
             rc -> patchState(arg("cid", rc), arg("wfid", rc), rc.request()))
     );
 
+    final List<Route<AsyncHandler<Response<ByteString>>>> forwardedRoutes = Arrays.asList(
+        Route.async(
+            "POST", BASE + "/<cid>",
+            rc -> createOrUpdateWorkflow(arg("cid", rc), rc)
+        ),
+        Route.async(
+            "DELETE", BASE + "/<cid>/<wfid>",
+            rc -> deleteWorkflow(arg("cid", rc), arg("wfid", rc), rc)
+        )
+    );
+
     return cat(
-        Api.prefixRoutes(routes, V3)
+        Api.prefixRoutes(routes, V3),
+        Api.prefixRoutes(forwardedRoutes, V3)
     );
   }
 
+  private CompletionStage<Response<ByteString>> deleteWorkflow(String cid, String wfid,
+                                                               RequestContext rc) {
+    // TODO: handle workflow crud directly in api service instead of proxying to scheduler
+    return rc.requestScopedClient()
+        .send(rc.request().withUri(schedulerApiUrl("workflows", cid, wfid)));
+  }
+
+  private CompletionStage<Response<ByteString>> createOrUpdateWorkflow(String componentId,
+                                                                       RequestContext rc) {
+    final Optional<ByteString> payload = rc.request().payload();
+    if (!payload.isPresent()) {
+      return CompletableFuture.completedFuture(
+          Response.forStatus(Status.BAD_REQUEST.withReasonPhrase("Missing payload.")));
+    }
+    final WorkflowConfiguration workflowConfig;
+    try {
+      workflowConfig = OBJECT_MAPPER
+          .readValue(payload.get().toByteArray(), WorkflowConfiguration.class);
+    } catch (IOException e) {
+      return CompletableFuture.completedFuture(
+          Response.forStatus(Status.BAD_REQUEST.withReasonPhrase("Invalid payload.")));
+    }
+
+    final Workflow workflow = Workflow.create(componentId, workflowConfig);
+
+    final ByteString requestPayload;
+    try {
+      requestPayload = serialize(workflow);
+    } catch (JsonProcessingException e) {
+      return CompletableFuture.completedFuture(Response.forStatus(
+          Status.INTERNAL_SERVER_ERROR.withReasonPhrase("Failed to serialize proxy payload.")));
```

Comment:
Let's get away from the `proxy` since it is completely internal to how we implement it.

Reply:
lol how did this go through the first PR

Reply:
also why is it re-serializing instead of just passing on inputs...

Reply:
not sure. I thought we fixed this.